src/queries/article/access/secret.ts

```diff
@@ -2,17 +2,27 @@ import { CIRCLE_STATE } from 'common/enums'
 import { ArticleAccessToSecretResolver } from 'definitions'
 
 export const secret: ArticleAccessToSecretResolver = async (
-  { articleId },
+  { articleId, authorId },
   _,
-  { knex }
+  { viewer, knex }
 ) => {
+  if (!authorId || !viewer.id) {
+    return
+  }
+
+  // check viewer is owner
+  if (authorId !== viewer.id) {
+    return
```

Comment:
Shall we throw an error instead of returning null? Returning null might be cached, then when the author query the secret s/he gets null as well

Reply:
should be fine since caching is based on `viewer.id`?

Reply:
Fixed it.